ivadomed/random.py

```diff
@@ -0,0 +1,16 @@
+import torch
+import random
+import numpy as np
+
+def set_seed(seed=6, rank=0):
+    """
+    This function controls sources of randomness to aid reproducibility.
+    Args:
+        seed: int to initialize random number generator (RNG)
+        rank: int specifying the GPU id for training
+    Returns:
+        seeded RNG in the specified libraries/packages.
+    """
+    random.seed(seed + rank)
+    np.random.seed(seed + rank)
+    torch.manual_seed(seed + rank)
```

Comment:
Why do we need the `rank` parameter? I'm not sure we want the seed to be different if we change GPUs or train on multiple GPUs.

Reply:
We don't necessarily need it for the current implementation of ivadomed. However, it is a convention used for multi GPU training and I left it for future works. Also, it doesn't hurt anything as of now. We can remove it, if it violates ivadomed conventions.

Reply:
Great idea and initiative 👍. I think that is a great future supporting perspective on controlling the variability at the hardware level. 

The potential downside (and this is a hunch for now) is that if  we use something like Compute Canada, the rank information... may be very unspecified? Sort of an edge case for now. Beside that point...

For localized training with controlled hardware environment, I think this works pretty flawlessly and FORCE the ORIGINAL experimenter to really replicate the experiment controlling the specific GPU used. I think that is a great idea. 

However, this situation may prove very... challenging for non ORIGINAL experimenter. E.g. Student A in another lab trying to reproduce our work will probably not be able to use the exact same hardware and GPU rank as we have (in fact a known issue with our current unit tests are failing with GPU unit tests because config requested a non-existent Rank), and or forcibly using a GPU in a particular slot just because we used that slot. Or worse, if  we have a Rank 7 GPU in a 8 GPU cluster setup, the students will never be able to replicate the results with same config UNLESS they tweak that `set_seed` function to manually set the values (which is beyond what we normally  expect from novice users/students outside of our contributor circle). 

Therefore I think it is probably safer at this point to not included it in line with the original goal of this PR, which is the allow seed fixing. By including Rank, we may in scenarios inadvertently  forcing unreproducible behavior via different seed, which goes against what you hope to achieve. 

I do think this is a valuable idea that when multi GPU support comes alive, we should make this an optional setting potential for people to set it. But that is well beyond current scope in terms of implementation/complexity so I do not recommend implemented at the moment.

Reply:
I agree with the fact that `rank` is not needed at this point, and it is mainly a feature of multi GPU use cases. The reason I included it is for us not to come back to this when we start supporting multi GPU.
However, I can take it out at this point.